src/main/java/org/apache/fineract/ReportApi/ReportService.java

```diff
@@ -0,0 +1,64 @@
+package org.apache.fineract.ReportApi;
+
+import com.amazonaws.services.applicationautoscaling.model.ValidationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ReportService {
+
+    @Autowired
+    private ReportRepository reportRepository;
+
+    public List<ReportRequest> getAllReports() {
+        return reportRepository.findAll();
+    }
+
+    public ReportRequest getReportById(Long id) {
+        return reportRepository.findById(id)
+                .orElseThrow(() -> new ReportNotFoundException("Report not found with id: " + id));
+    }
+
+    public ReportRequest createReport(@Valid ReportRequest reportRequest) {
+
+        if (reportRequest.getReportName() == null || reportRequest.getReportName().isEmpty()) {
```

Comment:
Many important validations are missing, like the query field is not null, and the query has no data modification statement.

Reply:
Added the validations